pakyow-core/lib/core/middleware/req_path_normalizer.rb

```diff
@@ -0,0 +1,27 @@
+require 'core/call_context'
+
+module Pakyow
+  module Middleware
+    # Rack compatible middleware that normalize the path if contains '//',
+    # it replace '//' with '/' and issue a 301 redirect to the new path.
+    #
+    # @api public
+    class ReqPathNormalizer
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if env['PATH_INFO'].include? '//'
+          CallContext.new(env).redirect(normalized_path(env['PATH_INFO']), 301)
```

Comment:
@bryanp I'll give that a shot and get back to you.
Pairing is an awesome idea :+1:, I know that you are super busy but if we can arrange a quick session that would be great :)